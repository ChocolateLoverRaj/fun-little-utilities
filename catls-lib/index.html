<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>catls-lib — References</title>
	<meta name="description" content="">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="assets/css/main.css">
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="assets/js/search.js" data-base=".">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="index.html" class="title">catls-lib — References</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
							<input type="checkbox" id="tsd-filter-only-exported" />
							<label class="tsd-widget" for="tsd-filter-only-exported">Only exported</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="globals.html">Globals</a>
				</li>
			</ul>
			<h1> catls-lib — References</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<div class="tsd-panel tsd-typography">
				<a href="#catls-lib" id="catls-lib" style="color: inherit; text-decoration: none;">
					<h1>CatLs Lib</h1>
				</a>
				<p>Core APIs used by <a href="https://npmjs.com/package/catls"><code>catls</code></a></p>
				<a href="#license" id="license" style="color: inherit; text-decoration: none;">
					<h2>License</h2>
				</a>
				<p><a href="https://git.io/fxKXN">MIT</a> © <a href="https://github.com/KSXGitHub">Hoàng Văn Khải</a></p>
			</div>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class="globals  ">
						<a href="globals.html"><em>Globals</em></a>
					</li>
					<li class="label tsd-is-external">
						<span>Internals</span>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_enums_index_.html">"catls-<wbr>lib/enums/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_empty_arguments_index_.html">"catls-<wbr>lib/functions/empty-<wbr>arguments/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_executor_index_.html">"catls-<wbr>lib/functions/executor/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_index_.html">"catls-<wbr>lib/functions/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_main_index_.html">"catls-<wbr>lib/functions/main/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_relative_link_index_.html">"catls-<wbr>lib/functions/relative-<wbr>link/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_show_exec_data_index_.html">"catls-<wbr>lib/functions/show-<wbr>exec-<wbr>data/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_stat_info_index_.html">"catls-<wbr>lib/functions/stat-<wbr>info/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_string_to_number_index_.html">"catls-<wbr>lib/functions/string-<wbr>to-<wbr>number/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_symlink_routing_functions_index_.html">"catls-<wbr>lib/functions/symlink-<wbr>routing-<wbr>functions/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_unit_index_.html">"catls-<wbr>lib/functions/unit/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_functions_unknown_stat_type_index_.html">"catls-<wbr>lib/functions/unknown-<wbr>stat-<wbr>type/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_index_.html">"catls-<wbr>lib/index"</a>
					</li>
					<li class=" tsd-kind-external-module">
						<a href="modules/_catls_lib_types_index_.html">"catls-<wbr>lib/types/index"</a>
					</li>
					<li class="label tsd-is-external">
						<span>Externals</span>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_event_target_proxy_index_.html">"event-<wbr>target-<wbr>proxy/index"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_evented_stream_types_index_.html">"evented-<wbr>stream-<wbr>types/index"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_iterate_evented_stream_index_.html">"iterate-<wbr>evented-<wbr>stream/index"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_index_.html">"split-<wbr>shell-<wbr>buffer/index"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_create_.html">"split-<wbr>shell-<wbr>buffer/lib/create"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_elements_.html">"split-<wbr>shell-<wbr>buffer/lib/elements"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_error_classes_.html">"split-<wbr>shell-<wbr>buffer/lib/error-<wbr>classes"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_from_child_process_.html">"split-<wbr>shell-<wbr>buffer/lib/from-<wbr>child-<wbr>process"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_from_evented_stream_.html">"split-<wbr>shell-<wbr>buffer/lib/from-<wbr>evented-<wbr>stream"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_from_iterable_.html">"split-<wbr>shell-<wbr>buffer/lib/from-<wbr>iterable"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_from_iterable_stream_.html">"split-<wbr>shell-<wbr>buffer/lib/from-<wbr>iterable-<wbr>stream"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_from_string_.html">"split-<wbr>shell-<wbr>buffer/lib/from-<wbr>string"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_lines_.html">"split-<wbr>shell-<wbr>buffer/lib/lines"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_splitter_object_.html">"split-<wbr>shell-<wbr>buffer/lib/splitter-<wbr>object"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_to_string_.html">"split-<wbr>shell-<wbr>buffer/lib/to-<wbr>string"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_types_.html">"split-<wbr>shell-<wbr>buffer/lib/types"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_write_.html">"split-<wbr>shell-<wbr>buffer/lib/write"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_lib_writeln_.html">"split-<wbr>shell-<wbr>buffer/lib/writeln"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_utils_call_maybe_function_.html">"split-<wbr>shell-<wbr>buffer/utils/call-<wbr>maybe-<wbr>function"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_utils_digit_.html">"split-<wbr>shell-<wbr>buffer/utils/digit"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_utils_sequence_tests_.html">"split-<wbr>shell-<wbr>buffer/utils/sequence-<wbr>tests"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_utils_simple_flatten_.html">"split-<wbr>shell-<wbr>buffer/utils/simple-<wbr>flatten"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_utils_special_character_.html">"split-<wbr>shell-<wbr>buffer/utils/special-<wbr>character"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_utils_string_writable_.html">"split-<wbr>shell-<wbr>buffer/utils/string-<wbr>writable"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_split_shell_buffer_utils_types_.html">"split-<wbr>shell-<wbr>buffer/utils/types"</a>
					</li>
					<li class=" tsd-kind-external-module tsd-is-external">
						<a href="modules/_ts_await_event_index_.html">"ts-<wbr>await-<wbr>event/index"</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-module"><span class="tsd-kind-icon">Module</span></li>
				<li class="tsd-kind-object-literal"><span class="tsd-kind-icon">Object literal</span></li>
				<li class="tsd-kind-variable"><span class="tsd-kind-icon">Variable</span></li>
				<li class="tsd-kind-function"><span class="tsd-kind-icon">Function</span></li>
				<li class="tsd-kind-function tsd-has-type-parameter"><span class="tsd-kind-icon">Function with type parameter</span></li>
				<li class="tsd-kind-index-signature"><span class="tsd-kind-icon">Index signature</span></li>
				<li class="tsd-kind-type-alias"><span class="tsd-kind-icon">Type alias</span></li>
				<li class="tsd-kind-type-alias tsd-has-type-parameter"><span class="tsd-kind-icon">Type alias with type parameter</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-enum"><span class="tsd-kind-icon">Enumeration</span></li>
				<li class="tsd-kind-enum-member"><span class="tsd-kind-icon">Enumeration member</span></li>
				<li class="tsd-kind-property tsd-parent-kind-enum"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-enum"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-interface"><span class="tsd-kind-icon">Interface</span></li>
				<li class="tsd-kind-interface tsd-has-type-parameter"><span class="tsd-kind-icon">Interface with type parameter</span></li>
				<li class="tsd-kind-constructor tsd-parent-kind-interface"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-interface"><span class="tsd-kind-icon">Method</span></li>
				<li class="tsd-kind-index-signature tsd-parent-kind-interface"><span class="tsd-kind-icon">Index signature</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-class"><span class="tsd-kind-icon">Class</span></li>
				<li class="tsd-kind-class tsd-has-type-parameter"><span class="tsd-kind-icon">Class with type parameter</span></li>
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class"><span class="tsd-kind-icon">Accessor</span></li>
				<li class="tsd-kind-index-signature tsd-parent-kind-class"><span class="tsd-kind-icon">Index signature</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-call-signature tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="assets/js/main.js"></script>
<script>if (location.protocol == 'file:') document.write('<script src="assets/js/search.js"><' + '/script>');</script>
</body>
</html>