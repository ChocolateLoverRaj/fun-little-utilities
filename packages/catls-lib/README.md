## CatLs Lib

Core APIs used by [`catls`](https://npmjs.com/package/catls)

## License

[MIT](https://git.io/fxKXN) © [Hoàng Văn Khải](https://github.com/KSXGitHub)
