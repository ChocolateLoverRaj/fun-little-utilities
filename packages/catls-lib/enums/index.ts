// tslint:disable:no-unnecessary-qualifier

export enum ExitStatus {
  Success = 0,
  InsufficientArguments = 1,
  NoEnt = 2,
  UncaughtException = 3
}

export enum EmptyArgumentHandlingMethod {
  Quiet = 'quiet',
  Warn = 'warn',
  Error = 'error'
}

export enum UnitType {
  Exception,
  Symlink,
  File,
  Directory,
  Unknown = -1
}

export namespace UnitType {
  export type Exist = UnitType.Symlink | UnitType.File | UnitType.Directory | UnitType.Unknown
  export type Known = UnitType.Exception | UnitType.Symlink | UnitType.File | UnitType.Directory
}

export enum SymlinkResolution {
  Agnostic = 'agnostic',
  Relative = 'relative',
  Ultimate = 'ultimate'
}

export enum UnknownStatTypeName {
  BlockDevice = 'BlockDevice',
  CharacterDevice = 'CharacterDevice',
  FIFO = 'FIFO',
  Socket = 'Socket',
  Unknown = 'Unknown'
}
