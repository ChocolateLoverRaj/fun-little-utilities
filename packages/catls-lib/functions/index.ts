import executor from './executor'
import main from './main'
import numberToString from './number-to-string'
import relativeLink from './relative-link'
import showExecData from './show-exec-data'
import statInfo from './stat-info'
import symlinkRoutingFunctions from './symlink-routing-functions'
import unit from './unit'
import unknownStatType from './unknown-stat-type'

export {
  executor,
  main,
  numberToString,
  relativeLink,
  showExecData,
  statInfo,
  symlinkRoutingFunctions,
  unit,
  unknownStatType
}
