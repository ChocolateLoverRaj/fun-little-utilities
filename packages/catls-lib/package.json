{
  "name": "catls-lib",
  "version": "0.0.6",
  "description": "Core APIs used by catls",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodeapps/fun-little-utilities#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodeapps/fun-little-utilities.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodeapps/fun-little-utilities/issues"
  },
  "keywords": [
    "cat",
    "ls"
  ],
  "dependencies": {
    "shell-quote": "^1.6.1",
    "typescript-miscellaneous": "^0.1.1",
    "convenient-typescript-utilities": "^0.0.3",
    "tslib": "^1.9.3",
    "split-shell-buffer": "^1.0.0-rc.2",
    "ts-await-event": "^0.0.0",
    "@types/shell-quote": "^1.6.0",
    "@types/node": "^10.12.18"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
