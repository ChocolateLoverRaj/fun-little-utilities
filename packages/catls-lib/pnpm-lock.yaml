dependencies:
  '@types/node': 11.11.1
  '@types/shell-quote': 1.6.1
  convenient-typescript-utilities: 0.0.3
  shell-quote: 1.6.1
  split-shell-buffer: 'link:../split-shell-buffer'
  ts-await-event: 'link:../ts-await-event'
  tslib: 1.9.3
  typescript-miscellaneous: 0.1.1
lockfileVersion: 5
packages:
  /@types/node/10.12.30:
    dev: false
    resolution:
      integrity: sha512-nsqTN6zUcm9xtdJiM9OvOJ5EF0kOI8f1Zuug27O/rgtxCRJHGqncSWfCMZUP852dCKPsDsYXGvBhxfRjDBkF5Q==
  /@types/node/11.11.1:
    dev: false
    resolution:
      integrity: sha512-2azXFP9n4aA2QNLkKm/F9pzKxgYj1SMawZ5Eh9iC21RH3XNcFsivLVU2NhpMgQm7YobSByvIol4c42ZFusXFHQ==
  /@types/shell-quote/1.6.1:
    dev: false
    resolution:
      integrity: sha512-t0bxRLQ75MMF7EeICa1eYC//o1/6gPaUV7ELke4l4OkwpZ9apOzvv2oR5F2PmQJ3tM83Lo+MNKfAXn5gQRMcnA==
  /array-filter/0.0.1:
    dev: false
    resolution:
      integrity: sha1-fajPLiZijtcygDWB/SH2fKzS7uw=
  /array-map/0.0.0:
    dev: false
    resolution:
      integrity: sha1-iKK6tz0c97zVwbEYoAP2b2ZfpmI=
  /array-reduce/0.0.0:
    dev: false
    resolution:
      integrity: sha1-FziZ0//Rx9k4PkR5Ul2+J4yrXys=
  /convenient-typescript-utilities/0.0.3:
    dependencies:
      '@types/node': 10.12.30
      tslib: 1.9.3
    dev: false
    engines:
      node: '>= 8.9.0'
    resolution:
      integrity: sha512-kHjMz9faySaKMx9ZRwv9oTogYRXCGaKPYq+xsLEswBo+0a5r8l5Xk/7Q+vqDLMYMrTm2oW7wEgtiDcXonr8Nkg==
  /jsonify/0.0.0:
    dev: false
    resolution:
      integrity: sha1-LHS27kHZPKUbe1qu6PUDYx0lKnM=
  /shell-quote/1.6.1:
    dependencies:
      array-filter: 0.0.1
      array-map: 0.0.0
      array-reduce: 0.0.0
      jsonify: 0.0.0
    dev: false
    resolution:
      integrity: sha1-9HgZSczkAmlxJ0MOo7PFR29IF2c=
  /tslib/1.9.3:
    dev: false
    resolution:
      integrity: sha512-4krF8scpejhaOgqzBEcGM7yDIEfi0/8+8zDRZhNZZ2kjmHJ4hv3zCbQWxoJGz1iw5U0Jl0nma13xzHXcncMavQ==
  /typescript-miscellaneous/0.1.1:
    dependencies:
      '@types/node': 10.12.30
      tslib: 1.9.3
    dev: false
    engines:
      node: '>= 8.9.0'
      typescript: '>= 3.0.0'
    resolution:
      integrity: sha512-dyGMilZPOL/qQMlJSWK+SbQyJpf6YZZPmPrbbdHGkAvGUvlhv4kGKTgddXWLxFWHTirR2I0VfD8mSwia3WLGtA==
specifiers:
  '@types/node': ^11.11.1
  '@types/shell-quote': ^1.6.1
  convenient-typescript-utilities: ^0.0.3
  shell-quote: ^1.6.1
  split-shell-buffer: ^1.0.0-rc.2
  ts-await-event: ^0.0.0
  tslib: ^1.9.3
  typescript-miscellaneous: ^0.1.1
