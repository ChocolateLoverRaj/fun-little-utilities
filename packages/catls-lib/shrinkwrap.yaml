dependencies:
  '@types/fs-extra': 5.0.4
  '@types/node': 10.12.18
  '@types/shell-escape': 0.2.0
  convenient-typescript-utilities: 0.0.3
  fs-extra: 7.0.1
  shell-escape: 0.2.0
  split-shell-buffer: 'link:../split-shell-buffer'
  ts-await-event: 'link:../ts-await-event'
  tslib: 1.9.3
  typescript-miscellaneous: 0.1.1
packages:
  /@types/fs-extra/5.0.4:
    dependencies:
      '@types/node': 10.12.10
    dev: false
    resolution:
      integrity: sha512-DsknoBvD8s+RFfSGjmERJ7ZOP1HI0UZRA3FSI+Zakhrc/Gy26YQsLI+m5V5DHxroHRJqCDLKJp7Hixn8zyaF7g==
  /@types/node/10.12.10:
    dev: false
    resolution:
      integrity: sha512-8xZEYckCbUVgK8Eg7lf5Iy4COKJ5uXlnIOnePN0WUwSQggy9tolM+tDJf7wMOnT/JT/W9xDYIaYggt3mRV2O5w==
  /@types/node/10.12.18:
    dev: false
    resolution:
      integrity: sha512-fh+pAqt4xRzPfqA6eh3Z2y6fyZavRIumvjhaCL753+TVkGKGhpPeyrJG2JftD0T9q4GF00KjefsQ+PQNDdWQaQ==
  /@types/shell-escape/0.2.0:
    dev: false
    resolution:
      integrity: sha512-7kUdtJtUylvyISJbe9FMcvMTjRdP0EvNDO1WbT0lT22k/IPBiPRTpmWaKu5HTWLCGLQRWVHrzVHZktTDvvR23g==
  /convenient-typescript-utilities/0.0.3:
    dependencies:
      '@types/node': 10.12.10
      tslib: 1.9.3
    dev: false
    engines:
      node: '>= 8.9.0'
    resolution:
      integrity: sha512-kHjMz9faySaKMx9ZRwv9oTogYRXCGaKPYq+xsLEswBo+0a5r8l5Xk/7Q+vqDLMYMrTm2oW7wEgtiDcXonr8Nkg==
  /fs-extra/7.0.1:
    dependencies:
      graceful-fs: 4.1.15
      jsonfile: 4.0.0
      universalify: 0.1.2
    dev: false
    engines:
      node: '>=6 <7 || >=8'
    resolution:
      integrity: sha512-YJDaCJZEnBmcbw13fvdAM9AwNOJwOzrE4pqMqBq5nFiEqXUqHwlK4B+3pUw6JNvfSPtX05xFHtYy/1ni01eGCw==
  /graceful-fs/4.1.15:
    dev: false
    resolution:
      integrity: sha512-6uHUhOPEBgQ24HM+r6b/QwWfZq+yiFcipKFrOFiBEnWdy5sdzYoi+pJeQaPI5qOLRFqWmAXUPQNsielzdLoecA==
  /jsonfile/4.0.0:
    dev: false
    optionalDependencies:
      graceful-fs: 4.1.15
    resolution:
      integrity: sha1-h3Gq4HmbZAdrdmQPygWPnBDjPss=
  /shell-escape/0.2.0:
    dev: false
    resolution:
      integrity: sha1-aP0CXrBJC09WegJ/C/IkgLX4QTM=
  /tslib/1.9.3:
    dev: false
    resolution:
      integrity: sha512-4krF8scpejhaOgqzBEcGM7yDIEfi0/8+8zDRZhNZZ2kjmHJ4hv3zCbQWxoJGz1iw5U0Jl0nma13xzHXcncMavQ==
  /typescript-miscellaneous/0.1.1:
    dependencies:
      '@types/node': 10.12.10
      tslib: 1.9.3
    dev: false
    engines:
      node: '>= 8.9.0'
      typescript: '>= 3.0.0'
    resolution:
      integrity: sha512-dyGMilZPOL/qQMlJSWK+SbQyJpf6YZZPmPrbbdHGkAvGUvlhv4kGKTgddXWLxFWHTirR2I0VfD8mSwia3WLGtA==
  /universalify/0.1.2:
    dev: false
    engines:
      node: '>= 4.0.0'
    resolution:
      integrity: sha512-rBJeI5CXAlmy1pV+617WB9J63U6XcazHHF2f2dbJix4XzpUF0RS3Zbj0FGIOCAva5P/d/GBOYaACQ1w+0azUkg==
registry: 'https://registry.npmjs.org/'
shrinkwrapMinorVersion: 9
shrinkwrapVersion: 3
specifiers:
  '@types/fs-extra': ^5.0.4
  '@types/node': ^10.12.18
  '@types/shell-escape': ^0.2.0
  convenient-typescript-utilities: ^0.0.3
  fs-extra: ^7.0.1
  shell-escape: ^0.2.0
  split-shell-buffer: ^1.0.0-rc.2
  ts-await-event: ^0.0.0
  tslib: ^1.9.3
  typescript-miscellaneous: ^0.1.1
