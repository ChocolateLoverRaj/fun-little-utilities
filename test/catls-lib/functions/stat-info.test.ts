import * as xjest from 'extra-jest'
import { statInfo, StatInfo } from 'catls-lib'
import UTCDate from '../.lib/utc-date'

const STATS: StatInfo.Stats = {
  atime: new UTCDate(1999, 2, 3, 1, 2, 3, 789),
  ctime: new UTCDate(2005, 3, 4, 2, 6, 4, 222),
  mtime: new UTCDate(1234, 5, 6, 7, 8, 9, 354),
  mode: 12,
  size: 34
}

const TYPE = '<Insert Type Here>'

describe('when body argument is not provided', () => {
  it('uses [] as body argument', () => {
    expect(statInfo(TYPE, STATS)).toEqual(statInfo(TYPE, STATS, []))
  })
})

it('matches snapshot', xjest.snap.default(
  statInfo(
    TYPE,
    STATS,
    [
      ['body.foo', 'body.bar'],
      ['body.baz', 'body.qux']
    ]
  ))
)
