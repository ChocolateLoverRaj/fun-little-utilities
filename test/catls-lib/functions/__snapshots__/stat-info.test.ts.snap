// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
"
|+
  [INFO]
    Type: <Insert Type Here>
    body.foo: body.bar
    body.baz: body.qux
    Size: 34
    Mode: 12
    Modified: 1234-06-06T07:08:09.354Z
    Accessed: 1999-03-03T01:02:03.789Z
    Changed: 2005-04-04T02:06:04.222Z


"
`;
